deep_qa/training/text_trainer.py

```diff
@@ -485,6 +484,20 @@ def _set_padding_lengths_from_model(self):
     # but simple models might be able to get by with just the default implementation
     ########################
 
+    def _get_instance_sorting_keys(self) -> List[str]:  # pylint: disable=no-self-use
+        """
+        If we're using dynamic padding, we want to group the instances by padding length, so that
+        we minimize the amount of padding necessary per batch.  This variable sets what exactly
+        gets sorted by.  We'll call
+        :func:`~deep_qa.data.instances.instance.IndexedInstance.get_padding_lengths()` on each
+        instance, pull out these keys, and sort by them in the order specified.  You'll want to
+        override this in your model class if you have more complex models.
+
+        The default implementation is to sort first by ``num_sentence_words``, then by
+        ``num_word_characters``.
+        """
+        return ['num_sentence_words', 'num_word_characters']
```

Comment:
I think this should be just `['num_sentence_words']` - in `instance.get_padding_lengths()` we assume that these keys exist in the length dictionary, but `num_word_characters` only exists if we are also using character embeddings (I think).